src/client/Microsoft.Identity.Client/Cache/CacheRefreshReason.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Specifies the reason for fetching the access token from the identity provider.
+    /// </summary>
+    public enum CacheRefreshReason
+    {
+        /// <summary>
+        /// When the cache is not supposed to be hit to make the request (interactive call, username password call, device code flow, etc.)
```

Comment:
nit: can we list all the flows / methods that hit the cache (so AcquireTokenSilent, AcquireTokenForClient and AcquireTokenOnBehalfOf)